contracts/0.8.9/WithdrawalQueue.sol

```diff
@@ -0,0 +1,368 @@
+    bytes32 public constant FINALIZE_ROLE = keccak256("FINALIZE_ROLE");
+    bytes32 public constant BUNKER_MODE_REPORT_ROLE = keccak256("BUNKER_MODE_REPORT_ROLE");
+
+    /// @notice minimal possible sum that is possible to withdraw
+    uint256 public constant MIN_STETH_WITHDRAWAL_AMOUNT = 100;
+
+    /// @notice maximum possible sum that is possible to withdraw by a single request
+    /// Prevents accumulating too much funds per single request fulfillment in the future.
+    /// @dev To withdraw larger amounts, it's recommended to split it to several requests
+    uint256 public constant MAX_STETH_WITHDRAWAL_AMOUNT = 1000 * 1e18;
+
+    /// @notice Lido stETH token address to be set upon construction
+    IStETH public immutable STETH;
+    /// @notice Lido wstETH token address to be set upon construction
+    IWstETH public immutable WSTETH;
+
+    /// @notice Emitted when the contract initialized
+    /// @param _admin provided admin address
+    /// @param _caller initialization `msg.sender`
+    event InitializedV1(address _admin, address _pauser, address _resumer, address _finalizer, address _caller);
+
+    error AdminZeroAddress();
+    error AlreadyInitialized();
+    error Uninitialized();
+    error Unimplemented();
+    error RequestAmountTooSmall(uint256 _amountOfStETH);
+    error RequestAmountTooLarge(uint256 _amountOfStETH);
+    error InvalidReportTimestamp();
+    error LengthsMismatch(uint256 _expectedLength, uint256 _actualLength);
+    error RequestIdsNotSorted();
+
+    /// @param _wstETH address of WstETH contract
+    constructor(IWstETH _wstETH) {
+        // init immutables
+        WSTETH = _wstETH;
+        STETH = WSTETH.stETH();
+    }
+
+    /// @notice Initialize the contract storage explicitly.
+    /// @param _admin admin address that can change every role.
+    /// @param _pauser address that will be able to pause the withdrawals
+    /// @param _resumer address that will be able to resume the withdrawals after pause
+    /// @param _finalizer address that can finalize requests in the queue
+    /// @dev NB! It's initialized in paused state by default and should be resumed explicitly to start
+    function initialize(address _admin, address _pauser, address _resumer, address _finalizer) external {
+        if (_admin == address(0)) revert AdminZeroAddress();
+
+        _initialize(_admin, _pauser, _resumer, _finalizer);
+    }
+
+    /// @notice Resume withdrawal requests placement and finalization
+    function resume() external whenPaused onlyRole(RESUME_ROLE) {
+        _resume();
+    }
+
+    /// @notice Pause withdrawal requests placement and finalization. Claiming finalized requests will still be available
+    /// @param _duration pause duration, seconds (use `PAUSE_INFINITELY` for unlimited)
+    function pause(uint256 _duration) external onlyRole(PAUSE_ROLE) {
+        _pause(_duration);
+    }
+
+    struct WithdrawalRequestInput {
```

Comment:
let's remove structure

Reply:
there is only single owner even in case of batching processing - all requests are belong to one user